contrib/epee/include/storages/portable_storage.h

```diff
@@ -241,40 +240,44 @@ namespace epee
     }
     //---------------------------------------------------------------------------------------------------------------
     template<class t_value>
-    bool portable_storage::set_value(const std::string& value_name, const t_value& v, hsection hparent_section)        
+    bool portable_storage::set_value(const boost::string_ref& value_name, const t_value& v, hsection hparent_section)        
     {
       BOOST_MPL_ASSERT(( boost::mpl::contains<boost::mpl::push_front<storage_entry::types, storage_entry>::type, t_value> )); 
       TRY_ENTRY();
       if(!hparent_section)
         hparent_section = &m_root;
-      storage_entry* pentry = find_storage_entry(value_name, hparent_section);
+      storage_entry* pentry = find_or_create_storage_entry(value_name, hparent_section);
       if(!pentry)
-      {
-        pentry = insert_new_entry_get_storage_entry(value_name, hparent_section, v);
-        if(!pentry)
-          return false;
-        return true;
-      }
+        return false;
       *pentry = storage_entry(v);
       return true;
       CATCH_ENTRY("portable_storage::template<>set_value", false);
     }
     //---------------------------------------------------------------------------------------------------------------
     inline
-    storage_entry* portable_storage::find_storage_entry(const std::string& pentry_name, hsection psection)
+    storage_entry* portable_storage::find_storage_entry(const boost::string_ref& pentry_name, hsection psection)
     {
       TRY_ENTRY();
       CHECK_AND_ASSERT(psection, nullptr);
-      auto it = psection->m_entries.find(pentry_name);
+      auto it = psection->m_entries.find(std::string(pentry_name.data(), pentry_name.size()));
       if(it == psection->m_entries.end())
         return nullptr;
 
       return &it->second;
       CATCH_ENTRY("portable_storage::find_storage_entry", nullptr);
     }
     //---------------------------------------------------------------------------------------------------------------
+    inline
+    storage_entry* portable_storage::find_or_create_storage_entry(const boost::string_ref& pentry_name, hsection psection)
+    {
+      TRY_ENTRY();
+      CHECK_AND_ASSERT(psection, nullptr);
+      return &psection->m_entries[std::string(pentry_name.data(), pentry_name.size())];
```

Comment:
This is why the entire interface makes no sense. A `backed_string_ref` was created to remove an allocation inside of `match_string2`, yet the changes are such that it will _always_ allocate a `std::string` right here or via `take_string()` in all other cases. There are no scenarios where the result of `match_string2` does not allocate a `std::string`, making the interface harder to use with no current benefit.